File: src/typescript/transaction/example-usage/example-usage/example-legacy-usage.ts
import { createHiveChain } from '@hiveio/wax';

// Initialize chain
const chain = await createHiveChain();

// Create/get a wallet
const { wallet, publicKey1 } = globalThis.snippetsBeekeeperData;

// Create a transaction
const tx = await chain.createTransaction();

// Declare example operation
const operation = {
  vote_operation: {
    voter: "voter",
    author: "author",
    permlink: "test-permlink",
    weight: 2200
  }
};

// Push the operation into the transaction
tx.pushOperation(operation);

// Convert the transaction into the Hive API-legacy form JSON before signing
const legacyApiTx = tx.toLegacyApi();

console.log(legacyApiTx);

// Because we want to process transction signing in legacy way, we need to sign the transaction externally, which is shown below.
// We need to calculate the transaction digest first.
const digest = tx.legacy_sigDigest;

/*
Other signers (except beekeeper) do not allow signing the digest directly,
this is a beekeeper-specific feature.
*/

// Generate the signature based on the transction digest
const signature = wallet.signDigest(publicKey1, digest);

// Suplement the transaction by created signature
tx.addSignature(signature);

// This is JSON form ready for broadcasting or passing to third-party service.
const txApiForm = tx.toLegacyApi();

console.log(txApiForm);

File: src/rust/transaction/example_usage/example_usage/example_legacy_usage.rs
use runner::{SnippetsBeekeeperData, snippets_beekeeper_data};
use wax::prelude::*;

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    // Initialize chain
    let chain = create_hive_chain(None)?;

    // Create/get a wallet
    let SnippetsBeekeeperData { mut wallet, public_key1, .. } = snippets_beekeeper_data();

    // Create a transaction
    let mut tx = chain.create_transaction(None).await?;

    // Declare example operation
    let operation = chain.create_operation(Value::VoteOperation(Vote {
        voter: "voter".into(),
        author: "author".into(),
        permlink: "test-permlink".into(),
        weight: 2200,
    }));

    // Push the operation into the transaction
    tx.push_operation(operation);

    // Convert the transaction into the Hive API-legacy form JSON before signing
    let legacy_api_tx = tx.to_legacy_api()?;

    println!("{legacy_api_tx}");

    // Because we want to process transaction signing in legacy way, we need to sign the transaction externally, which is shown below.
    // We need to calculate the transaction digest first.
    let digest = tx.legacy_sig_digest()?;

    /*
    Other signers (except beekeeper) do not allow signing the digest directly,
    this is a beekeeper-specific feature.
    */

    // Generate the signature based on the transaction digest
    let signature = wallet.sign_digest(&public_key1, &digest)?;

    // Supplement the transaction by created signature
    tx.add_signature(&signature)?;

    // This is JSON form ready for broadcasting or passing to third-party service.
    let tx_api_form = tx.to_legacy_api()?;

    println!("{tx_api_form}");

    Ok(())
}
